Gems/EMotionFX/Code/EMotionFX/Source/Recorder.cpp

```diff
@@ -1421,17 +1421,16 @@ namespace EMotionFX
     // extract sorted active items
     void Recorder::ExtractNodeHistoryItems(const ActorInstanceData& actorInstanceData, float timeValue, bool sort, EValueType valueType, AZStd::vector<ExtractedNodeHistoryItem>* outItems, AZStd::vector<size_t>* outMap) const
     {
-        // clear the map array
+        // Reinit the item array.
         const size_t maxIndex = CalcMaxNodeHistoryTrackIndex(actorInstanceData);
         outItems->resize(maxIndex + 1);
         for (size_t i = 0; i <= maxIndex; ++i)
```

Comment:
Might as well use a range-for loop here instead, then delete line 1429:
```suggestion
        for (ExtractedNodeHistoryItem& item : *outItems)
```